x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_diff_tab.tsx

```diff
@@ -16,22 +16,125 @@ import {
+  */
+  'actions',
+  'exceptions_list',
+
+  /*
+    Most prebuilt rules are installed as "disabled". Once user enables a rule, it'll show as diff, which doesn't add value.
+  */
+  'enabled',
+
+  /* Technical property. Hidden because it can be confused with "version" by users. */
+  'revision',
+
+  /*
+    "updated_at" value is regenerated on every '/upgrade/_review' endpoint run 
+    and will therefore always show a diff. It adds no value to display it to the user.
+  */
+  'updated_at',
+];
 
 const sortAndStringifyJson = (jsObject: Record<string, unknown>): string =>
   stringify(jsObject, { space: 2 });
 
+/**
+ * Normalizes the rule object, making it suitable for comparison with another normalized rule.
+ *
+ * Normalization is needed to avoid showing diffs for semantically equal property values.
+ * Saving a rule via the rule editing UI might change the format of some properties or set default values.
+ * This function compensates for those changes.
+ *
+ * @param {RuleResponse} originalRule - Rule of a RuleResponse type.
+ * @returns {RuleResponse} - The updated normalized rule object.
+ */
+const normalizeRule = (originalRule: RuleResponse): RuleResponse => {
+  const rule = { ...originalRule };
+
+  /*
+    Convert the "from" property value to a humanized duration string, like 'now-1m' or 'now-2h'. 
+    Conversion is needed to skip showing the diff for the "from" property when the same 
+    duration is represented in different time units. For instance, 'now-1h' and 'now-3600s' 
+    indicate a one-hour duration.
+    The same helper is used in the rule editing UI to format "from" before submitting the edits. 
+    So, after the rule is saved, the "from" property unit/value might differ from what's in the package.
+  */
+  rule.from = formatScheduleStepData({
+    interval: rule.interval,
+    from: getHumanizedDuration(rule.from, rule.interval),
+    to: rule.to,
+  }).from;
+
+  /*
+    Default "note" to an empty string if it's not present.
+    Sometimes, in a new version of a rule, the "note" value equals an empty string, while 
+    in the old version, it wasn't specified at all (undefined becomes ''). In this case, 
+    it doesn't make sense to show diff, so we default falsy values to ''.
+  */
+  rule.note = rule.note ?? '';
+
+  /*
+    Removes empty arrays (techniques, subtechniques) from the MITRE ATT&CK value.
+    The same processing is done in the rule editing UI before submitting the edits.
+  */
+  rule.threat = filterEmptyThreats(rule.threat);
+
+  /*
+    The "machine_learning_job_id" property is converted from the legacy string format 
+    to the new array format during installation and upgrade. Thus, all installed rules 
+    use the new format. For correct comparison, we must ensure that the new rule is 
+    also in the new format before showing the diff.
```

Comment:
Not sure I understand this one. Shouldn't this be: 

For correct comparison, we must ensure that the **old** rule is  also in the new format before showing the diff.
?

And the usecase that I imagine is a user having a really old prebuilt rule installed, which is still in the old format, and it is compared against the new format from the target rule?
Where does this migration take place?

Reply:
Yeah, after re-reading I agree that the wording is confusing. It should be
> For correct comparison, we must ensure that the **rule update** is also in the new format...


My idea was to avoid this situation. Let's say, there's a rule that uses the old format ([example](https://github.com/elastic/integrations/blob/4029af97a8c345176860e5df0a6b64bcf4df2731/packages/security_detection_engine/kibana/security_rule/19de8096-e2b0-4bd8-80c9-34a820813fff_105.json#L14)). User installs it and before installation we pass the asset through `normalizeMachineLearningJobIds` which converts a string into array. Then Elastic releases an update for this rule and it also uses the string format. Now we'll need to pass the update to `normalizeMachineLearningJobIds` to be able to compare array with array.

Regarding your use case, it should also be fine, since both the installed rule and the update are transformed by this `normalizeRule` function, so both will end up being arrays. I hope it makes sense :)

Reply:
Ok, I understand. But I guess my doubt came from the fact that I supposed that if Elastic releases an update to a rule it will have the `machine_learning_job_id` property in the new format, not in the legacy one. 

So sure, the safe way to go here is to normalize both old rule and new rule into the new format direction and compare them so. 👍

Reply:
Thx! Updated the code comment.